lib/controls/ValueEditor.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+class ValueEditor extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    const {field, value, operator, handleOnChange} = this.props;
+    const children = this.props.children;
+
+    if (operator === 'null' || operator === 'notNull') {
+        return null;
+    }
+
+    if (children) {
+      let type = children.type;
+      if (type.shouldRender && type.shouldRender(this.props)) {
```

Comment:
We could just use React.createElement and pass in the props. Let the custom component decide whether or not to render